### PR TITLE
custom template for iterative

### PR DESCRIPTION
**Pull request by skshetry**
@iterative/dvc 

Tree view: https://github.com/iterative/py-template/tree/custom-template

# How to test:
```console
# setup
git clone git@github.com:iterative/py-template.git
git checkout custom-template
pip install cookiecutter
# creating templates:
cookiecutter .
```

Uses:
- nox for tests automation/lints
- `setuptools` as backend, build as frontend
- ...

TODO:
- [x] Code coverage
- [x] Contributing
- [x] Code of conduct
- [x] Nox vs tox?
- [ ] Using poetry?
- [ ] Docs

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,13 +1,38 @@
...
-    rev: v2.3.0
+  - repo: https://github.com/codespell-project/codespell
+    rev: v2.1.0
+    hooks:
+      - id: codespell
+  - repo: https://github.com/asottile/pyupgrade
+    rev: v2.31.0
+    hooks:
+      - id: pyupgrade
+  - repo: https://github.com/timothycrosley/isort
```
Comment (casperdcl) [inline .pre-commit-config.yaml:23]:
```suggestion
  - repo: https://github.com/PyCQA/isort
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,13 +1,38 @@
...
+  - repo: https://github.com/asottile/pyupgrade
+    rev: v2.31.0
+    hooks:
+      - id: pyupgrade
+  - repo: https://github.com/PyCQA/isort
+    rev: 5.10.1
+    hooks:
+      - id: isort
+        language_version: python3
+  - repo: https://gitlab.com/pycqa/flake8
```
Comment (casperdcl) [inline .pre-commit-config.yaml:28]:
```suggestion
  - repo: https://github.com/pre-commit/mirrors-mypy
    rev: v0.930
    hooks:
      - id: mypy
  - repo: https://gitlab.com/pycqa/flake8
```

**Review by casperdcl — CHANGES_REQUESTED** [review]
some pre-commit upgrades

File: .pre-commit-config.yaml (hunk 2 above)
Comment (skshetry) [reply .pre-commit-config.yaml:28]: We don't use mypy through pre-commit, as it's slow and requires access to local dependencies. Thus we test them via `nox`.

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -1,13 +1,38 @@
...
     hooks:
-      - id: prettier
+      - id: flake8
+        language_version: python3
+        additional_dependencies:
+          - flake8-bugbear
+          - flake8-comprehensions
+          - flake8-debugger
+          - flake8-string-format
+          - flake8-bandit
```
[the remark this answers is not in the document]
Comment (skshetry) [reply .pre-commit-config.yaml:38]: We have `mypy` for annotations, and we have `isort` above.
```suggestion
          - flake8-bandit
          - flake8-broken-line
          - flake8-bugbear
          - flake8-comprehensions
          - flake8-debugger
          - flake8-string-format
```

**Pull request merged**